presto-main/src/test/java/com/facebook/presto/operator/scalar/TestStringFunctions.java

```diff
@@ -380,6 +380,102 @@ public void testTrim()
+
+        // non latin characters
+        assertFunction("LTRIM('źółć', 'óź')", createVarcharType(4), "łć");
+
+        // invalid utf-8 characters
+        assertFunction("CAST(LTRIM(utf8(from_hex('81')), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81));
+        assertFunction("CAST(LTRIM(CONCAT(utf8(from_hex('81')), ' '), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81, ' '));
+        assertFunction("CAST(LTRIM(CONCAT(' ', utf8(from_hex('81'))), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81));
+        assertFunction("CAST(LTRIM(CONCAT(' ', utf8(from_hex('81')), ' '), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81, ' '));
+        assertInvalidFunction("LTRIM('źółć', utf8(from_hex('81')))", createVarcharType(4), "Invalid UTF-8 encoding");
+        assertInvalidFunction("LTRIM('źółć', utf8(from_hex('3281')))", createVarcharType(4), "Invalid UTF-8 encoding");
+    }
+
+    private SqlVarbinary varbinary(int... bytesAsInts)
+    {
+        byte[] bytes = new byte[bytesAsInts.length];
+        for (int i = 0; i < bytes.length; i++) {
+            bytes[i] = (byte) bytesAsInts[i];
+        }
+        return new SqlVarbinary(bytes);
+    }
+
+    @Test
+    public void testRightTrimParametrized()
+    {
+        assertFunction("RTRIM('', '')", createVarcharType(0), "");
+        assertFunction("RTRIM('   ', '')", createVarcharType(3), "   ");
+        assertFunction("RTRIM('  hello  ', '')", createVarcharType(9), "  hello  ");
+        assertFunction("RTRIM('  hello  ', ' ')", createVarcharType(9), "  hello");
+        assertFunction("RTRIM('  hello  ', 'lo ')", createVarcharType(9), "  he");
+        assertFunction("RTRIM('hello  ', ' ')", createVarcharType(7), "hello");
+        assertFunction("RTRIM('hello  ', 'l o')", createVarcharType(7), "he");
+        assertFunction("RTRIM('hello  ', 'l')", createVarcharType(7), "hello  ");
+        assertFunction("RTRIM(' hello world ', ' ')", createVarcharType(13), " hello world");
+        assertFunction("RTRIM(' hello world ', ' ld')", createVarcharType(13), " hello wor");
+        assertFunction("RTRIM(' hello world ', ' ehlowrd')", createVarcharType(13), "");
+        assertFunction("RTRIM(' hello world ', ' x')", createVarcharType(13), " hello world");
+
+        // non latin characters
+        assertFunction("RTRIM('źółć', 'ćł')", createVarcharType(4), "źó");
+
+        // invalid utf-8 characters
+        assertFunction("CAST(RTRIM(utf8(from_hex('81')), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81));
+        assertFunction("CAST(RTRIM(CONCAT(utf8(from_hex('81')), ' '), ' ') AS VARBINARY)", VARBINARY, varbinary(0x81));
+        assertFunction("CAST(RTRIM(CONCAT(' ', utf8(from_hex('81'))), ' ') AS VARBINARY)", VARBINARY, varbinary(' ', 0x81));
+        assertFunction("CAST(RTRIM(CONCAT(' ', utf8(from_hex('81')), ' '), ' ') AS VARBINARY)", VARBINARY, varbinary(' ', 0x81));
+        assertInvalidFunction("RTRIM('źółć', utf8(from_hex('81')))", createVarcharType(4), "Invalid UTF-8 encoding");
```

Comment:
When you're testing invalid second argument, it doesn't really matter what you provide in the first argument. I'll just use something simple (instead of non-ASCII characters)